File: examples/primes.r
IsPrime <- function(n) {
    for (i in 2:(n/2)) {
        if ( n %% i == 0) {
            return(FALSE)
        }
    }
    return(TRUE)
}

for (i in 2:100)
    if (IsPrime(i))
        print(i)

File: examples/primes.cxx

#include <iostream>

bool IsPrime(int n)
{
    for ( int i = 2; i <= n / 2; ++i) {
        if (n % i == 0) {
           return false;
        }
    }
 return true;
}

int main()
{
    for (int i = 2; i< 100000; i++)
    {
        if (IsPrime(i))
           std::cout <<i<<"\n";
    }
}